src/sql-parser/src/parser.rs

```diff
@@ -6032,115 +6032,179 @@ impl<'a> Parser<'a> {
+            None => self.parse_revoke_role(),
+        }
+    }
+
+    /// Parse a `REVOKE PRIVILEGE` statement, assuming that the `REVOKE` token
+    /// and all privileges have already been consumed.
+    fn parse_revoke_privilege(
+        &mut self,
+        privileges: PrivilegeSpecification,
+    ) -> Result<Statement<Raw>, ParserError> {
+        self.expect_keyword(ON)?;
+        let objects = if self.parse_keyword(ALL) {
+            let object_type = self.expect_grant_revoke_plural_object_type_in_schema("REVOKE")?;
+            self.expect_keywords(&[IN, SCHEMA])?;
+            let schemas = self.parse_comma_separated(Parser::parse_schema_name)?;
+            GrantObjectSpecification {
+                object_type,
+                object_spec_inner: GrantObjectSpecificationInner::All { schemas },
+            }
+        } else {
+            let object_type = self.expect_grant_revoke_object_type("REVOKE")?;
+            let names =
+                self.parse_comma_separated(|parser| parser.parse_object_name(object_type))?;
+            GrantObjectSpecification {
+                object_type,
+                object_spec_inner: GrantObjectSpecificationInner::Objects { names },
+            }
+        };
+        self.expect_keyword(FROM)?;
+        let roles = self.parse_comma_separated(Parser::expect_role_specification)?;
+        Ok(Statement::RevokePrivileges(RevokePrivilegesStatement {
+            privileges,
+            objects,
+            roles,
+        }))
+    }
+
+    /// Parse a `REVOKE ROLE` statement, assuming that the `REVOKE` token
+    /// has already been consumed.
+    fn parse_revoke_role(&mut self) -> Result<Statement<Raw>, ParserError> {
+        let role_name = self.parse_identifier()?;
+        self.expect_keyword(FROM)?;
+        let member_names = self.parse_comma_separated(Parser::expect_role_specification)?;
+        Ok(Statement::RevokeRole(RevokeRoleStatement {
+            role_name,
+            member_names,
+        }))
+    }
+
+    /// Bail out if the current token is not an object type suitable for a GRANT/REVOKE, or consume
+    /// and return it if it is.
+    fn expect_grant_revoke_object_type(
+        &mut self,
+        statement_type: &str,
+    ) -> Result<ObjectType, ParserError> {
+        // If the object type is omitted, then it is assumed to be a table.
+        let object_type = self.parse_object_type().unwrap_or(ObjectType::Table);
+        self.expect_grant_revoke_object_type_inner(statement_type, object_type)
+    }
+
+    /// Bail out if the current token is not a plural object type suitable for a GRANT/REVOKE, or consume
+    /// and return it if it is.
+    fn expect_grant_revoke_plural_object_type_in_schema(
+        &mut self,
+        statement_type: &str,
+    ) -> Result<ObjectType, ParserError> {
+        let object_type = self.expect_plural_object_type().map_err(|_| {
+            // Limit the error message to allowed object types.
```

Comment:
I don't understand this. Why are the 4 below the only allowed object types?

Reply:
Those are the only valid object types that exist in a schema. All relations are covered by `TABLES`.

Reply:
The table rule is so weird.

Reply:
Agreed, it's completely inconsistent with every other command.